lib/ddtrace/contrib/http/instrumentation.rb

```diff
@@ -29,11 +29,13 @@ def self.after_request(&block)
         end
 
         # InstanceMethods - implementing instrumentation
-        module InstanceMethods
+        module InstanceMethods # rubocop:disable Metrics/ModuleLength:
           include Datadog::Contrib::HttpAnnotationHelper
 
           # :yield: +response+
           def request(req, body = nil, &block)
+            return super(req, body, &block) unless started?
```

Comment:
The internal implementation makes starts the connection and call `request` in recursion if the connection is not yet started.

This would mean we would create two request spans for every request call. This guard make sure it does not happen - it pass to super when connection is not started. This will do what is described above, and will pass to `request` method again, this time with started connection.